src/test/java/com/ning/http/client/async/NonBlockingInputStreamFeederTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package com.ning.http.client.async;
+
+import static java.lang.System.arraycopy;
+import static java.util.Arrays.copyOf;
+import static java.util.Arrays.copyOfRange;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyBoolean;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.spy;
+import static org.testng.Assert.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.glassfish.grizzly.Buffer;
+import org.testng.annotations.Test;
+
+import com.ning.http.client.providers.grizzly.FeedableBodyGenerator;
+import com.ning.http.client.providers.grizzly.NonBlockingInputStreamFeeder;
+
+public class NonBlockingInputStreamFeederTest {
+
+  private static final byte[] DATA = "Hello, world!".getBytes();
+  private static final byte[] EMPTY = new byte[0];
+
+  @Test
+  public void feedSmallPayload() throws IOException {
+    assertPayload(DATA);
+  }
+
+  @Test
+  public void feedEmptyPayload() throws IOException {
+    assertPayload(EMPTY);
+  }
+
+  protected void assertPayload(byte[] data) throws IOException {
```

Comment:
why protected?

Reply:
In case a subclass wants to reuse this assertion method.  Should I make it private?

Reply:
I think this should be private for the moment.

Reply:
I think it should be private for the moment.